RecoEgamma/EgammaElectronAlgos/src/GsfElectronAlgo.cc

```diff
@@ -692,6 +710,12 @@ void GsfElectronAlgo::calculateShowerShape_full5x5( const reco::SuperClusterRef
     showerShape.hcalTowersBehindClusters = generalData_->hcalHelper->hcalTowersBehindClusters(*theClus) ;
     showerShape.hcalDepth1OverEcalBc = generalData_->hcalHelper->hcalESumDepth1BehindClusters(showerShape.hcalTowersBehindClusters)/showerShape.e5x5 ;
     showerShape.hcalDepth2OverEcalBc = generalData_->hcalHelper->hcalESumDepth2BehindClusters(showerShape.hcalTowersBehindClusters)/showerShape.e5x5 ;
+    // FIXME this should get its own flag, but nevermind for now
+    if (showerShape.hcalDepth1OverEcal == 0 && showerShape.hcalDepth2OverEcal == 0) {
+      if (!generalData_->hcalHelper->hasActiveHcal(*theClus)) {
+        showerShape.hcalDepth1OverEcal = showerShape.hcalDepth2OverEcal = -1e-6; // set to negative value
+      }
+    }
```

Comment:
thank you for doing it for both fractional and full5x5 showershapes so that they are the same and we dont end up with confusion as a result